### PR TITLE
add options for target triple and cpu (-mtriple and -mcpu) for LLVM

### PR DESCRIPTION
**Pull request by caoimhinuibrian**
Signed-off-by: Kevin O'Brien <caomhin@us.ibm.com>

### CODE REVIEW

**Comment by caoimhinuibrian** [comment]
This is an example:
Command -

root@00b70efeab49:~/workspace# onnx-mlir --EmitLib -mtriple=s390x-ibm-loz -mcpu=z15 cos_op.onnx

Results in:

llc: error: : error: unable to get target for 's390x-ibm-loz -unknown-mcpu=z15', see --version and --triple.
llc -filetype=obj -relocation-model=pic --mtriple=s390x-ibm-loz --mcpu=z15 -o cos_op.o cos_op.bc

This is expected because we build LLVM with only host backend support, so on my mac the only targets are x86 triples and cpus.

File: src/MainUtils.cpp [hunk 1]
```diff
@@ -288,6 +317,15 @@ void genConstPackObj(const mlir::OwningModuleRef &module,
...
 
+string getTargetOptions() {
+  string targetOptions = "";
+  if (mtriple != "")
+    targetOptions = "--mtriple=" + mtriple;
+  if (mcpu != "")
+    targetOptions += " --mcpu=" + mcpu;
+  return targetOptions;
+}
+
```
Comment (AlexandreEichenberger) [inline src/MainUtils.cpp:328]: It turns out that the same triple/cpu option does work for both the O3 and llc, so its great that the same option is passed for both.

File: src/MainUtils.cpp [hunk 2]
```diff
@@ -321,6 +360,7 @@ void genModelObject(const mlir::OwningModuleRef &module, string bitcodePath,
   Command llvmToObj(/*exePath=*/!llcPath.empty() ? llcPath : kLlcPath);
   llvmToObj.appendStr("-filetype=obj")
       .appendStr("-relocation-model=pic")
+      .appendStr(getTargetOptions())
```
Comment (AlexandreEichenberger) [inline src/MainUtils.cpp:363]: Any reasons that we also add "-O3" here?

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, if you don't mind, I think we should also add the -O3 to llc, which does not appears to be on at this time.

Tx

File: src/MainUtils.cpp (hunk 2 above)
Comment (caoimhinuibrian) [reply src/MainUtils.cpp:363]: the pipeline is to call opt to optimize the llvm IR (with -O3) and subsequently call llc to take the optimized bitcode and convert to machine code .... so, we don't need -O3 on the llc patrt

**Pull request merged**